### PR TITLE
Live time tracking

### PR DESCRIPTION
**Pull request by sergej-tschak**

### CODE REVIEW

**Review by benj56 — DISMISSED** [review]

**Review by AlGvrl — APPROVED** [review]

File: src/absences/index.ts [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+import responseHandler from '../utils/responseHandlers';
+import { ReadOptions } from '../utils/types';
+import { AxiosRequestConfig } from 'axios';
 
 export default class Absences extends BaseApi {
   static resourceName = 'absences';
 
-  public read(): Promise<Absence[]> {
-    const response = this.get<Absence[]>(`${Absences.resourceName}/read`);
+  public read(options: ReadOptions = {}): Promise<Absence[]> {
```
Comment (tomi-timetac) [inline src/absences/index.ts:10]: Should be readAll to be consistent with other endpoints.

File: src/timetrackings/index.ts [hunk 2]
```diff
@@ -1,36 +1,44 @@
...
+import { AxiosRequestConfig } from 'axios';
+import { ReadOptions } from '../utils/types';
 
 export default class TimeTrackings extends BaseApi {
-  static resourceName = 'absences';
+  static resourceName = 'timeTrackings';
 
-  public read(): Promise<TimeTracking[]> {
-    const response = this.get<TimeTracking>(`${TimeTrackings.resourceName}/read`);
+  read(id: number, options: ReadOptions = {}) {
```
Comment (tomi-timetac) [inline src/timetrackings/index.ts:10]: Maybe a better name for options would be params, as options can't be set. I would also change the ReadOptions to ReadParams. And options/params should be optional. Also in readAll.

**Review by AlGvrl — DISMISSED** [review]

**Review by tomi-timetac — CHANGES_REQUESTED** [review]

**Review by benj56 — DISMISSED** [review]

File: src/absences/index.ts (hunk 1 above)
Comment (AlGvrl) [reply src/absences/index.ts:10]: I thought we switched from `readAll`s to `read`s in #2 ?

Comment (tomi-timetac) [reply src/absences/index.ts:10]: Rename single fetch (read/{id}) to **readById** (returns only one entity) and **readAll** should be just **read** with options/params (returns array of entities).

**Review by tomi-timetac — APPROVED** [review]

**Pull request merged**